doc/src/ch03-04-creating-our-source-part-3-processing-events-almost.md

```diff
@@ -0,0 +1,161 @@
+# Creating our source, part III: processing events (almost)
+
+Finally, the real functionality we care about! Processing events! This is also a method in the `calloop::EventSource` trait:
+
+```rust,noplayground
+fn process_events<F>(
+    &mut self,
+    readiness: calloop::Readiness,
+    token: calloop::Token,
+    mut callback: F,
+) -> io::Result<calloop::PostAction>
+where
+    F: FnMut(Self::Event, &mut Self::Metadata) -> Self::Ret,
+```
+
+What a mouthful! But when you break it down, it's not so complicated:
+
+- We take our own state, of course, as `&mut self`.
+
+- We take a `Readiness` value - this is mainly useful for "real" file descriptors, and tells you whether the event source was woken up for a read or write event. We ignore it though, because our internal sources are always only readable (remember that even if the zsocket is writeable, the FD it exposes is only ever readable).
+
+- We take a token, just like our register/re-register methods. The ID will always correspond to our own source, but we can check the sub-ID to see which of our internal sources caused us to run.
```

Comment:
The `sub_id` is no longer visible, you updated the example accordingly, but this sentence is now outdated.

Reply:
Good catch, thanks.